aiida/orm/nodes/caching.py

```diff
@@ -138,8 +138,8 @@ def _iter_all_same_nodes(self, allow_before_store=False) -> t.Iterator['Node']:
         builder.append(self._node.__class__, filters={f'extras.{self._HASH_EXTRA_KEY}': node_hash}, subclassing=False)
 
         return (
-            node for node in builder.all(flat=True) if node.base.caching.is_valid_cache
-        )  # type: ignore[misc,union-attr]
+            node for node, in builder.iterall() if node.base.caching.is_valid_cache  # type: ignore[misc,union-attr]
```

Comment:
would it make sense for `iterall` to also have a `flat` argument?

Reply:
There is an open issue for this https://github.com/aiidateam/aiida-core/issues/5982 But [as I mentioned there](https://github.com/aiidateam/aiida-core/issues/5982#issuecomment-1521944187) I remember looking into it when implementing it for `all` and `first` and found it non-trivial. I can have another look at some point and if I manage, will go through the code to use it where possible.